Controls/Grid/Grid.UWP/View/Services/Selection/SelectionService.cs

```diff
@@ -540,11 +540,18 @@ private void SelectRangeUnits(int row, int startColumnIndex, int endColumnIndex)
 
         private void SelectRangeCells(int row, int column = -1)
         {
-            var cells = this.Owner.Model.CellsController.GetCellsForRow(row);
-            GridCellModel model = column != -1 ? cells.FirstOrDefault(a => a.Column.ItemInfo.Slot == column) : cells.FirstOrDefault();
-            if (model != null)
+            var item = this.Owner.GetDataView().Items[row];
```

Comment:
Cache the "GetDataView". Every-time when GetDataView gets executed a new instance of "DataGridDataView" gets created. Lets cache the GetDataView inside a field that should be initialized initially inside the constructor of the service.

Reply:
The cached value of the data view should probably be updated in `OnAttached()`/`OnDetached()` since `Owner` is a public property that could potentially be changed after construction.

Reply:
The same might be said for `selectedItems.SelectionUnit`. However, I suppose it's possible in practice that `Owner` never changes after construction. I'm not familiar enough with this code base to be certain though.

Reply:
Good job. I think that part of my concerns is resolved.